src/map/utils/fishingcontest.cpp

```diff
@@ -0,0 +1,716 @@
+        }
+
+        // Fill out the entry with the current data
+        entry->mjob        = (uint8)PChar->GetMJob();
+        entry->sjob        = (uint8)PChar->GetSJob();
+        entry->mlvl        = PChar->GetMLevel();
+        entry->slvl        = PChar->GetSLevel();
+        entry->race        = PChar->mainlook.race;
+        entry->allegiance  = (uint8)PChar->allegiance;
+        entry->fishRank    = PChar->RealSkills.rank[SKILLTYPE::SKILL_FISHING];
+        entry->score       = score;
+        entry->submitTime  = CVanaTime::getInstance()->getVanaTime();
+        entry->contestRank = 0;
+        entry->share       = 0;
+        entry->dataset_b   = 0;
+
+        // Update the database
+        WriteContestEntryData(entry);
+    }
+
+    void WithdrawFish(CCharEntity* PChar)
+    {
+        FishingContestEntry* entry = GetPlayerEntry(PChar);
+        if (entry)
+        {
+            // Remove from the database
+            const char* Query = "DELETE FROM `fishing_contest_entries` \
+                                 WHERE `name` = '%s';";
+            int32       ret   = _sql->Query(Query, PChar->getName());
+
+            if (ret == SQL_ERROR)
+            {
+                ShowError("Failed to remove entry from fishing entry database.");
+                return;
+            }
+
+            // Remove from the in-memory vector
+            // clang-format off
+            auto it = std::find_if(FishingContestEntries.begin(), FishingContestEntries.end(),
+                                [&PChar](FishingContestEntry* e) -> bool
+                                {
+                                    return strcmp(PChar->name.c_str(), e->name) == 0;
+                                });
+            // clang-format on
+
+            if (it != FishingContestEntries.end())
+            {
+                int index = it - FishingContestEntries.begin();
+                FishingContestEntries.erase(FishingContestEntries.begin() + index);
+            }
+        }
+    }
+
+    void ClaimContestReward(CCharEntity* PChar)
+    {
+        if (PChar == nullptr)
+        {
+            ShowDebug("Attempted to issue contest reward to a null pointer character.");
+            return;
+        }
+
+        if (CurrentFishingContest.status != CONTEST_STATUS_PRESENTING)
+        {
+            ShowDebug("Attempted to issue reward to player [%s] outside of 'Presenting' status.", PChar->getName());
+            return;
+        }
+
+        const char* Query = "UPDATE fishing_contest_entries "
+                            "SET claimed = 1 "
+                            "WHERE name = '%s';";
+
+        int32 ret = _sql->Query(Query, PChar->getName());
+        if (ret == SQL_ERROR)
+        {
+            ShowDebug("Unable to remove award on fishing contest for player %s.", PChar->getName());
+        }
+    }
+
+    auto GetFishingContestHistory(CCharEntity* PChar) -> std::tuple<uint16, uint16, uint16, uint16>
```

Comment:
Can this tuple be a named struct? This is very anonymous

Reply:
absolutely -- I only did it this way because it's only used once to return the rank 1 / rank 2 / rank 3 / rank 4-10 values.  I'll build a struct!